packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -61,78 +77,171 @@ export default class DropdownCore extends React.Component<DropdownCoreProps> {
     }
 
     componentDidMount() {
-        document.addEventListener("mouseup", this._handleInteract);
-        document.addEventListener("touchend", this._handleInteract);
-        document.addEventListener("keyup", this._handleKeyup);
+        this.updateEventListeners();
+    }
+
+    componentDidUpdate(prevProps: DropdownCoreProps) {
+        if (prevProps.open !== this.props.open) {
+            this.updateEventListeners();
+        }
+
+        // If the menu just changed from closed to open, and if we haven't
+        // found the correct menuWidth yet
+        if (!prevProps.open && this.props.open && !this.menuWidth) {
+            // There is a timeout because if we call getBoundingClientRect
+            // right away, it returns the value of the document width instead
+            // of the width of the just-mounted component, the menu.
+            setTimeout(() => {
```

Comment:
🤷‍♂️ I think doing `requestAnimationFrame` might be the preferred method of running something on the next render frame (and I think it's what Aphrodite does, too).